sinatra-contrib/lib/sinatra/content_for.rb

```diff
@@ -128,9 +128,12 @@ module ContentFor
     #
     # Your blocks can also receive values, which are passed to them
     # by <tt>yield_content</tt>
-    def content_for(key, value = nil, &block)
+    def content_for(key, value = nil, options = {}, &block)
+      key = key.to_sym
+
       block ||= proc { |*| value }
-      content_blocks[key.to_sym] << capture_later(&block)
+      clear_content_for(key) if options[:flush]
```

Comment:
@iguchi1124 this is the only line that needs added in this method. Can you rebase with master and limit changes to just this line and the added tests?

Reply:
Just tried to resolve the conflicts by web GUI.